product_supplierinfo_for_customer/models/product_product.py

```diff
@@ -64,34 +64,23 @@ def _get_price_from_customerinfo(self, partner_id):
         self.ensure_one()
         if not partner_id:
             return 0.0
-        customerinfo = self.env["product.customerinfo"].search(
-            [
-                ("name", "=", partner_id),
-                "|",
-                ("product_id", "=", self.id),
-                "&",
-                ("product_tmpl_id", "=", self.product_tmpl_id.id),
-                ("product_id", "=", False),
-            ],
-            limit=1,
-            order="product_id, sequence",
-        )
+        if partner_id and isinstance(partner_id, int):
```

Comment:
`partner_id` shouldn't be polymorphic and only receives an integer IMO.

Reply:
can you make this concession at least? this won't affect in any way any installation...

Reply:
Yes, it does. Any subsequent inherited method will need to check the same each time.

Reply:
That's why type hinting was added in Python 3.x, for avoiding these ugly hacks.

Reply:
Ok, what should I put then?

Reply:
```
...
partner = self.env["res.partner"].browse(partner_id)
customerinfo = self._select_customerinfo(partner=partner)
```

will this be ok?

Reply:
pushed, hope it is good now

Reply:
As the argument is the integer, always pass the ID, and do the browse if you need the recordset. On a migration to next non migrated version, the argument may be changed to `partner`.

Reply:
it is done, could you check?